libevm/lib/service_state_refund.go

```diff
@@ -0,0 +1,61 @@
+package lib
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	gethParams "github.com/ethereum/go-ethereum/params"
+)
+
+// setStateRefunds replicates the refund part of the original SSTORE gas consumption logic:
```

Comment:
It seems that we implement `EIP-1283` here and it is the most actual implementation, right?
Let's add the regression test coverage using the test data from it: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1283.md#test-cases

I can see similar for the eip2200 in Geth code - `eip2200Tests`.
It is strange that they have only legacy tests there...

Reply:
Not quite, 1283 was the first modification to SSTORE refunds, then came 2200, 2929 and then 3529. This implements the latest one, 3529. I will add their test cases from [EIP-3529](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3529.md#test-cases).

Reply:
Ok, thanks. than 3529

Reply:
done